src/states/node.rs

```diff
@@ -721,7 +720,7 @@ impl Node {
                    self,
                    peer_id,
                    hop_msg);
-            return Err(RoutingError::UnknownConnection(peer_id));
+            *self.name()
```

Comment:
I agree with Diggory here - it's very unclear to me why we choose ourselves as the hop when the sender is unknown to us.